externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/ui/KyuubiHistoryServerPlugin.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.ui
+
+import org.apache.spark.SparkConf
+import org.apache.spark.kyuubi.SparkSQLEngineEventListener
+import org.apache.spark.scheduler.SparkListener
+import org.apache.spark.status.{AppHistoryServerPlugin, ElementTrackingStore}
+
+import org.apache.kyuubi.config.KyuubiConf
+import org.apache.kyuubi.engine.spark.events.EngineEventsStore
+
+class KyuubiHistoryServerPlugin extends AppHistoryServerPlugin {
+
+  private val kyuubiConf: KyuubiConf = KyuubiConf()
+
+  override def createListeners(conf: SparkConf, store: ElementTrackingStore): Seq[SparkListener] = {
+    addKyuubiConfFromSpark(conf)
+    Seq(new SparkSQLEngineEventListener(store, kyuubiConf))
+  }
+
+  override def setupUI(ui: SparkUI): Unit = {
+    addKyuubiConfFromSpark(ui.conf)
```

Comment:
it looks like we should build kyuubi conf for every Spark app UI instead of use a global kyuubi conf.

Reply:
> it looks like we should build kyuubi conf for every Spark app UI instead of use a global kyuubi conf.

These sparkConf seem to be the same object, should we create multiple kyuubiConf?

Reply:
If we need to get the configuration of different spark app, we can get it using `org.apache.spark.status.AppStatusStore#environmentInfo`, do we need to do that?

Reply:
> These sparkConf seem to be the same object, should we create multiple kyuubiConf?

I just check the Spark build-in plugin, do not see a global value. The code is confused if the input conf is always same. It's better to craete multiple kyuubiConf.

> If we need to get the configuration of different spark app, we can get it using org.apache.spark.status.AppStatusStore#environmentInfo, do we need to do that?

It seems trivial, we dont need to update.